portfolio/src/main/webapp/script.js

```diff
@@ -23,10 +23,80 @@ function addRandomGreeting() {
   const greeting = greetings[Math.floor(Math.random() * greetings.length)];
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
+  const greetingContainer = document.getElementById('greetingContainer');
   greetingContainer.innerText = greeting;
 }
 
+const images = [
+  {name: 'basketball.jpg', label: 'Playing basketball with the Tennis Intramural Team!'},
+  {name: 'chess.jpg', label: 'Playing chess with my roommate!'},
+  {name: 'crochet.jpg', label: 'Crocheting simple beanies!'},
+  {name: 'smoothie.jpg', label: 'Making delicious smoothies!'},
+  {name: 'statefarm.jpg', label: 'State Farm Internship!'},
+  {name: 'tennis.jpg', label: 'Tennis, with the Club Team at UT Austin!'},
+  {name: 'UIL_state.jpg', label: 'Competitive Programming!'},
+  {name: 'Uke.jpg', label: 'Playing the Ukulele!'}
+]
+
+const DIRECTION_UP = 'up';
+const DIRECTION_DOWN = 'down';
+const RANDOM_SLIDE = 'random';
+
+let slideShowIndex = 0;
+
+/* Navigate to the slide at slideShowIndex - 1 */
+function previousSlide() {
+  updateSlideShowIndex(DIRECTION_DOWN);
+  let image = images[slideShowIndex];
+  displayNewSlide(image);
+}
+
+/* Navigate to the slide at slideShowIndex + 1 */
+function nextSlide() {
+  updateSlideShowIndex(DIRECTION_UP);
+  let image = images[slideShowIndex];
+  displayNewSlide(image);
+}
+
+/* Navigate to a random slide */
+function randomSlide() {
+  updateSlideShowIndex(RANDOM_SLIDE);
+  let image = images[slideShowIndex];
+  displayNewSlide(image);
+}
+
+/* Display slideshow, with correctly navigated content */
+function displayNewSlide(image) {
+  // display slide/image on slideshow
+  const imgElement = document.createElement('img');
+  imgElement.src = 'images/' + image.name;
+  const imageContainer = document.getElementById('slideshowContentContainer');
+  imageContainer.innerHTML = '';
+  imageContainer.appendChild(imgElement);
+
+  // display corresponding description text
+  const paragraph = document.createElement('p');
+  paragraph.appendChild(document.createTextNode(image.label));
+  const descriptionContainer = document.getElementById('slideDescription');
+  descriptionContainer.innerHTML = '';
+  descriptionContainer.appendChild(paragraph);
+}
+
+/* Navigating the Slideshow with an index */
+function updateSlideShowIndex(direction) {
+  if (direction == DIRECTION_UP) {
+    slideShowIndex = (slideShowIndex + 1) % images.length;
+  }
+  else if (direction == DIRECTION_DOWN) {
+    slideShowIndex = slideShowIndex - 1;
+    if (slideShowIndex < 0) {
+      slideShowIndex += images.length;
+    }
+  }
+  else if (direction == RANDOM_SLIDE) {
+    slideShowIndex = Math.floor(Math.random() * images.length);
+  }
```

Comment:
nit: consider making this whole thing a `switch(){}` statement

Reply:
Made the change! The code looked more organized. Are switch statements a little bit faster?